components/react/hooks/src/useDynamicHeightCollapsible/index.js

```diff
@@ -0,0 +1,17 @@
+import {useState, useCallback} from 'react'
+import useToggle from '../useToggle'
+
+export default function useDynamicHeightCollapsible(getContentHeighFn) {
+  const [isCollapsed, toggleCollapsible] = useToggle(true)
+  const [contentHeight, setContentHeight] = useState(null)
+  const heightRecalculator = useCallback(() => {
```

Comment:
I think in this case adding the useCallback hook could be a deoptimization because of these reasons:
- The dependency is a function, which most of the times will be declared inline or will need to be memorized by the parent component, which will include more complexity.
- Memoizing this function makes sense if the main purpose is to pass its reference to an atomic component event or a memorized component, like a button for example. Is this the main scope of this function?
I think would be better to remove it until the use case of this function is not clear in different cases, we can always refactor it and add the memoization.

Reply:
done!